node/src/filecheckpointer.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2021 IBM All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import fs from 'fs';
+import { Checkpointer } from './checkpointer';
+
+export interface CheckpointerState {
+  blockNumber?: string;
+  transactionIDs: string[];
+}
+
+export class FileCheckPointer implements Checkpointer {
+    #path: string;
+    #blockNumber?: bigint;
+    #transactionIDs: string[]=[]
+
+    constructor(path: string) {
+        this.#path = path;
+    }
+
+    async init(): Promise<void> {
+        await this.#getStateFromFile();
+        await this.#saveDate();
+    }
+
+    async checkpoint(blockNumber: bigint, transactionId?: string): Promise<void> {
+        if (blockNumber !== this.#blockNumber) {
+            this.#blockNumber = blockNumber;
+            this.#transactionIDs = [];
+        }
+        if (transactionId) {
+            this.#transactionIDs.push(transactionId);
+        }
+        await this.#saveDate();
+    }
+
+    getBlockNumber(): bigint | undefined {
+        return this.#blockNumber;
+    }
+
+    getTransactionIds(): string[] {
+        return this.#transactionIDs;
+    }
+
+    async #getStateFromFile(): Promise<void> {
+        if (fs.existsSync(this.#path)) {
```

Comment:
Sync I/O methods will block process execution so prefer the promise API instead. The exists check is actually not  very robust as the file still might not be there or be readable when the read call happens. Better to just do the read without an exists check, and catch the error if the read fails